src/librustc_metadata/encoder.rs

```diff
@@ -65,6 +66,20 @@ macro_rules! encoder_methods {
     }
 }
 
+/// A check to see if we are using a Yorick tracer. Used to decide whether we need to encode MIR
+/// for all items.
+fn building_with_yk_tracer() -> bool {
+    match env::var("YK_TRACER") {
+        Ok(val) => {
+            match val.as_str() {
+                "hw" | "sw" | "sw-rustc" => true,
```

Comment:
Why do we have `sw` and `sw-rustc`?

Reply:
It's explained here:
https://github.com/softdevteam/ykrustc/pull/41/files#diff-1aeb4fd1a8c45243dc9f0a52c1dafc36R8

In short, it's a mode which inserts calls to the trace recorder, but which doesn't encode SIR. We use this for compiling ykrustc itself (we never want to trace the compiler binaries, but we do want to trace it's libraries).

Reply:
OK, got it.

Reply:
In hindsight, I wonder if `sw-nosir` would have been a better name.

Reply:
I tend to agree that's a better name. We can just add a commit at the end to that effect: I don't think we need to edit all the existing commits.